### PR TITLE
feat: add icon components

### PR DESCRIPTION
**Pull request by claudio-OST**
This pull request introduces a set of icon components to our application's UI library.

Each icon component has been integrated into Storybook, allowing for easy visualization and testing. This ensures that the icons can be reviewed in isolation and in different scenarios, confirming their design and functionality.

For consistency and ease of use, default properties such as **size, color, and, where applicable, filled**, have been set for each icon. These defaults ensure a consistent look and feel across our application while allowing flexibility for specific use cases.

### CODE REVIEW

File: src/components/icons/iconUtils.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+		case 'base':
+			return 'fill-base-600';
+		case 'secondary':
+			return 'fill-secondary-600';
+		case 'primary':
+			return 'fill-primary-600'; 
+		case 'white':
+			return 'fill-white'; 
+		default:
+			return 'fill-slate-600'; // Default
```
Comment (malinovic) [inline src/components/icons/iconUtils.js:25]: The default comment is not necessary since the switch already describes that it is the default case.

**Review by malinovic — CHANGES_REQUESTED** [review]
Hi claudio, good job. There are a few things that I have noticed:

### JSX -> TSX

All files here are jsx files. Please change the file type to tsx since we want to have types also it has to be changed. Other than that, all icons are missing its "props" (see button). 

### Styling

I have seen that you set the transform property on the icon arrow. I would rather change it and only use tailwind classes for that.

### Storybook

The current structure that we have is rather confusing to me. For me it is not clear, why we have 3 sections (Icons Arrow, Icons Fillable, Icons simple). The difference in functionality is obvious, but I would rather put them into one section "Icons" and create a mdx description on how to utilise it. Otherwise the enduser would not have a clear direct understanding of it.

### Conclusion
So far I really like the technical implementation of the icons. Especially the separation of icons into it's own components so the user can only really select one icon directly. The utils file does also a very nice job. It is mostly only formal things, that I think could be a little bit different. 👍🏻

**Comment by malinovic** [comment]
Also one thing that I have noticed, the PR is not up to date with the main. Please also merge the main branch into your current branch so it is updated. Thank's. 😁

File: src/components/icons/iconUtils.js (hunk 1 above)
Comment (claudio-OST) [reply src/components/icons/iconUtils.js]: @malinovic fixed

**Review by malinovic — APPROVED** [review]
Great refactoring work, I like it! Approved!

**Pull request merged**